app/controllers/config_reports_controller.rb

```diff
@@ -49,4 +68,19 @@ def csv_columns
   def resource_base
     super.my_reports
   end
+
+  def can_delete?(report)
+    authorized_for(auth_object: report, authorizer: authorizer, permission: "destroy_#{controller_permission}")
+  end
+
+  def can_view?(report)
+    authorized_for(auth_object: report, authorizer: authorizer, permission: "view_#{controller_permission}")
+  end
+
+  def origin_icon(report)
+    origin = report.origin
+    return 'N/A' if origin.blank?
+    origin_icon = try("#{origin.downcase}_report_origin_icon".to_sym)
+    helpers.image_tag(origin_icon || origin + ".png", :title => _("Reported by %s") % origin, :width => "20px")
+  end
```

Comment:
Can the icon building be done in the UI instead and here only send a value `null/ansible/...`